### PR TITLE
Linux fixes for ncurses & readline.

### PR DESCRIPTION
**Pull request by qookei**

### CODE REVIEW

**Review by no92 — APPROVED** [review]

**Comment by Dennisbonke** [comment]
LGTM

**Review by Dennisbonke — APPROVED** [review]

File: sysdeps/linux/generic/sysdeps.cpp [hunk 1]
```diff
@@ -398,6 +408,35 @@ int sys_tgkill(int tgid, int tid, int sig) {
...
+}
+
+int sys_tcsetattr(int fd, int optional_action, struct termios *attr) {
+	int req;
+
+	switch (optional_action) {
+		case TCSANOW: req = TCSETS; break;
+		case TCSADRAIN: req = TCSETSW; break;
+		case TCSAFLUSH: req = TCSETSF; break;
+	}
```
Comment (avdgrinten) [inline sysdeps/linux/generic/sysdeps.cpp:425]: `default: return EINVAL;`? What do other libcs do here?

File: sysdeps/linux/generic/sysdeps.cpp [hunk 2]
```diff
@@ -398,6 +408,35 @@ int sys_tgkill(int tgid, int tid, int sig) {
 	return 0;
 }
 
+int sys_tcgetattr(int fd, struct termios *attr) {
+	auto ret = do_syscall(NR_ioctl, TCGETS, attr);
```
Comment (czapek1337) [inline sysdeps/linux/generic/sysdeps.cpp:412]: The `fd` is missing from the syscall arguments

File: sysdeps/linux/generic/sysdeps.cpp [hunk 3]
```diff
@@ -398,6 +408,35 @@ int sys_tgkill(int tgid, int tid, int sig) {
...
+int sys_tcsetattr(int fd, int optional_action, struct termios *attr) {
+	int req;
+
+	switch (optional_action) {
+		case TCSANOW: req = TCSETS; break;
+		case TCSADRAIN: req = TCSETSW; break;
+		case TCSAFLUSH: req = TCSETSF; break;
+	}
+
+	auto ret = do_syscall(NR_ioctl, req, attr);
```
Comment (czapek1337) [inline sysdeps/linux/generic/sysdeps.cpp:427]: Same here as well

**Review by czapek1337 — CHANGES_REQUESTED** [review]

File: sysdeps/linux/generic/sysdeps.cpp [hunk 4]
```diff
@@ -398,6 +408,35 @@ int sys_tgkill(int tgid, int tid, int sig) {
 	return 0;
 }
 
+int sys_tcgetattr(int fd, struct termios *attr) {
```
Comment (czapek1337) [inline sysdeps/linux/generic/sysdeps.cpp:411]: Also the `sys_tcgetattr` signature expects the `attr` pointer to be const

**Review by czapek1337 — CHANGES_REQUESTED** [review]

**Pull request merged**